pkg/apis/v1alpha1/nodeoverlay.go

```diff
@@ -32,7 +32,7 @@ type NodeOverlaySpec struct {
 	// - Custom labels from NodePool's spec.template.labels
 	// +kubebuilder:validation:XValidation:message="requirements with operator 'NotIn' must have a value defined",rule="self.all(x, x.operator == 'NotIn' ? x.values.size() != 0 : true)"
 	// +kubebuilder:validation:XValidation:message="requirements with operator 'In' must have a value defined",rule="self.all(x, x.operator == 'In' ? x.values.size() != 0 : true)"
-	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt' or 'Lt' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt') ? (x.values.size() == 1 && int(x.values[0]) >= 0) : true)"
+	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt', 'Lt', 'Gte', or 'Lte' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt' || x.operator == 'Gte' || x.operator == 'Lte') ? (x.values.size() == 1 && int(x.values[0]) >= 0) : true)"
 	// +kubebuilder:validation:MaxItems:=100
 	// +required
 	Requirements []v1.NodeSelectorRequirement `json:"requirements,omitempty"`
```

Comment:
Should we use a new type here rather than continuing to use the upstream type? It's not strictly necessary, but it strikes me as strange to allow values which wouldn't normally be valid for the upstream type. Encoding it in a different type makes it clear that it accepts all valid Karpenter operators, rather than all valid upstream operators.

Reply:
Misunderstood this to be a comment re: overflow and our validation. I get that the issue is specific to overlay. Fixing.